src/pages/index.js

```diff
@@ -0,0 +1,238 @@
+let section;
+api
+  .getInitialCards()
+  .then((items) => {
+    section = new Section(
+      {
+        items: items,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          return cardElement;
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+/* ---------- Get Profile ------------ */
+const userInfo = new UserInfo();
+api
+  .getUserInfo()
+  .then((profileData) => {
+    userInfo.setUserInfo(profileData);
+  })
+  .catch((err) => console.error(err));
+
+/* ---------- Patch Profile Edit Function ------------ */
+function handleProfileEditSubmit(profileInputValues) {
+  editProfileModal.setLoading(true);
+
+  const userData = {
+    name: profileInputValues.title,
+    about: profileInputValues.description,
+  };
+
+  api
+    .editUserInfo(userData.name, userData.about)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo(updatedUserData);
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false);
+    });
+}
+
+/* ---------- Patch Profile Avatar Function ------------ */
+function handleAvatarSubmit({ url }) {
+  updateAvatarModal.setLoading(true);
+
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(url);
+      updateAvatarModal.reset();
+      updateAvatarModal.close();
+      console.log("Success:", res);
+    })
+    .catch((err) => {
+      console.error("Error updating profile avatar:", err);
+    })
+    .finally(() => {
+      updateAvatarModal.setLoading(true, "Save");
+    });
+}
+
+/* ---------- Post Add Place Function ------------ */
+function handleNewPlaceSubmit(placeCardData) {
+  addPlaceModal.setLoading(true);
+  const cardData = {
+    name: placeCardData.title,
+    link: placeCardData.url,
+  };
+
+  api
+    .addNewPlace(cardData.name, cardData.link)
+    .then((newPlaceCard) => {
+      const cardElement = getCardElement({
+        name: newPlaceCard.name,
+        link: newPlaceCard.link,
+        _id: newPlaceCard._id,
+      });
+      section.addItem(cardElement);
+      addPlaceModal.close();
+      addPlaceModal.reset();
+    })
+    .catch((err) => {
+      console.error("Error adding new place:", err);
+    })
+    .finally(() => {
+      addPlaceModal.setLoading(false);
+      addPlaceValidation.disableButton();
+    });
+}
+
+/* ---------- Put Add Like React ------------ */
+function handleLikeReact(likeReact, likeStatus, cardId) {
+  if (likeStatus) {
+    api
+      .removeLikeReact(cardId)
+      .then(() => {
+        likeReact.classList.remove("card__react-button_active");
+      })
+      .catch((error) => console.error("Error removing like reaction:", error));
+  } else {
+    api
+      .addLikeReact(cardId)
+      .then(() => {
+        likeReact.classList.add("card__react-button_active");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle